listenbrainz/db/couchdb.py

```diff
@@ -0,0 +1,249 @@
+import json
+import re
+from typing import TextIO
+
+import requests
+import ujson
+from requests.adapters import HTTPAdapter
+from sentry_sdk import start_span
+
+from urllib3 import Retry
+
+# stat type followed by a underscore followed by a date in YYYYMMDD format
+DATABASE_NAME_PATTERN = re.compile(r"([a-zA-Z]+)_(\w+)_(\d{8})")
+
+DATABASE_LOCK_FILE = "LOCK"
+
+_user = None
```

Comment:
When I was working with this code for the spotify cache, I was wondering why you chose not to make this module a class...

Reply:
I do not remember distinctly, it may be that I forgot about trying that. How about I try moving this to a class in a follow up PR?

Reply:
Sounds good.